packages/mui-base/src/useButton/useButton.types.ts

```diff
@@ -22,6 +22,17 @@ export interface UseButtonParameters {
    * @default 'button'
    */
   type?: React.ButtonHTMLAttributes<HTMLButtonElement>['type'];
+  /**
+   * The HTML element, e.g.'button', 'a' etc
+   * @default ''
+   */
+  hostElementName?: keyof HTMLElementTagNameMap;
+  /**
+   * The name of the component using useButton.
+   * For debugging purposes.
+   * @default 'useButton'
+   */
+  componentName?: string;
```

Comment:
I'm not a fan of such internal props (especially when nothing other than the docs hint that they're internal), both here and in the Button component. Could we just use the word "Button" in error messages?